### PR TITLE
676 allow no regexp for parent in FASTA importer

### PR DESCRIPTION
**Pull request by bradfordcondon**
# New Feature

Issue #676 

## Description
The FASTA importer adds a relationship to another feature if you specify the type and a regexp.  however a regexp is not necessary if the parent feature name is identical to the feature you are importing.

This simple PR allows a form to be submitted with a parent feature type and relationship type but no regexp.  And, if this is so, it will search for the parent with the same uniquename.  This feature is found in other importers as discussed in #676.

## Testing?
As a bonus i wrote a test case for this feature.  We insert two mRNAs.  then we load polypeptides with the same names and verify that the importer links them (that the mRNA has a feature_relationship record)

### CODE REVIEW

**Comment by bradfordcondon** [comment]
ill look into the failed test today, it passes locally...


oops, i added a join and forgot you cant chain joins.  should be fixed.

File: tripal_chado/includes/TripalImporter/FASTAImporter.inc [hunk 1]
```diff
@@ -536,7 +535,7 @@ class FASTAImporter extends TripalImporter {
         }
 
         // Get the relationship subject
-         $subject = "";
+         $subject = $name ? $name : $uname;
```
Comment (bradfordcondon) [inline tripal_chado/includes/TripalImporter/FASTAImporter.inc]: should this actually be $uname or null? or instead of $name or $uname?

Comment (laceysanderson) [reply tripal_chado/includes/TripalImporter/FASTAImporter.inc]: I think it should be 
```suggestion
         $subject = $uname ? $uname : "";
```
based on https://github.com/tripal/tripal/pull/738/files#diff-8006ac719a97e82d92833b60886802a2L831 where the query looking up the parent/subject for the relationship checks the uniquename column.

**Review by laceysanderson — COMMENTED** [review]
Test passes locally for me 👍

**Review by laceysanderson — APPROVED** [review]

**Comment by bradfordcondon** [comment]
the second review for this PR does not need to be functional.

**Review by spficklin — APPROVED** [review]

**Comment by spficklin** [comment]
Non-functional code review looks good. I merged it.

**Pull request merged**